src/pages/signage/index.js

```diff
@@ -0,0 +1,274 @@
+    getEvents(newLocation, '', 1);
+    getBanners(newLocation, '', 1);
+    props.getSign(newLocation);
+  };
+  
+  const onSearch = (newTerm) => {
+    if (view === 'activities') {
+      getEvents(null, newTerm,1);
+    } else if (view === 'banners') {
+      getBanners(null, newTerm, 1);
+    }
+  };
+  
+  const onSort = (index, key, dir, func) => {
+    if (view === 'activities') {
+      getEvents(null, null,null, key, dir);
+    } else if (view === 'banners') {
+      getBanners(null, null, null, key, dir);
+    }
+  };
+  
+  const onPageChange = (newPage) => {
+    if (view === 'activities') {
+      getEvents(null, null, newPage);
+    } else if (view === 'banners') {
+      getBanners(null, null, newPage);
+    }
+  };
+  
+  const jumpToEvent = (id) => {
+    const event = events.find(ev => ev.id === id);
+    props.publishDate(event.start_date);
+  };
+  
+  const viewSign = () => {
+    window.open(`${window.SIGNAGE_BASE_URL}/${template}#/?summit=${summit.id}&location=${locationId}`);
+  };
+  
+  const pushUpdates = () => {
+    props.publishSignUpdates(template, jumpDate, staticBannerLoc);
+  };
+  
+  if(!summit.id || !locations) return(<div />);
+  
+  const eventsColumns = [
+    { columnKey: 'id', value: 'Id', sortable: true },
+    { columnKey: 'title', value: T.translate("signage.title"), sortable: true },
+    { columnKey: 'speakers_str', value: T.translate("signage.speakers")},
+    { columnKey: 'floor_loc', value: T.translate("signage.floor_loc")},
+    { columnKey: 'start_date_str', value: T.translate("signage.start_date"), sortable: true},
+    { columnKey: 'end_date_str', value: T.translate("signage.end_date")},
+  ];
+  
+  const eventsTableOptions = {
+    sortCol: props.order,
+    sortDir: props.orderDir,
+    actions: {
+      custom: [
+        {
+          name: 'jump',
+          tooltip: 'jump to this event',
+          icon: <i className="fa fa-share" />,
+          onClick: jumpToEvent,
+        }
+      ],
+    }
+  };
+  
+  return (
+    <div className="signage-wrapper">
+      <Breadcrumb data={{ title: T.translate("signage.signs"), pathname: match.url }} />
+      <div className="container">
+        <div className={styles.header}>
+          <div className="row form-group">
+            <div className="col-md-4">
+              <label> {T.translate("signage.location")} </label>
+              <LocationGroupedDropdown
+                locations={locations}
+                value={locationId}
+                placeholder={T.translate("signage.placeholders.locations")}
+                onChange={op => onChangeLocation(op.value)}
+              />
+            </div>
+            <div className="col-md-4">
+              <label> {T.translate("signage.template")} </label>
+              <Dropdown
+                placeholder={T.translate("signage.placeholders.template")}
+                value={template}
+                onChange={ev => setTemplate(ev.target.value)}
```

Comment:
set template needs to write the value to the api and publish the update to the ably channel as its does the original admin

Reply:
well its just setting the state the on change event as far i can see
the pushing  of the update its done when u hit the publish update button here 
https://github.com/fntechgit/summit-admin/blob/3afb8657f30f062f01ee710e335ce3fd7af03ff0/src/pages/signage/index.js#L195
and originally on the former admin UX was to publish the update on the onchange
check here
https://github.com/fntechgit/openstacksignage/blob/39a896a54034106a378f4831e8198de65b011698/admin.html#L1126